bitcoin/examples/taproot-psbt.rs

```diff
@@ -579,7 +580,10 @@ impl BenefactorWallet {
             let mut origins = BTreeMap::new();
             origins.insert(
                 beneficiary_key,
-                (vec![leaf_hash], (self.beneficiary_xpub.fingerprint(), new_derivation_path)),
+                (
+                    vec![leaf_hash],
+                    (self.beneficiary_xpub.fingerprint(), new_derivation_path.to_derivation_path()),
```

Comment:
In this case i dont see a good fix... `KeySource` is a tupple spread all of the code... trying any abstraction or change on it will run out of scope for this pr.

Reply:
I mean the conversion.

Reply:
I'll hack deeper for that

Reply:
'''Rust
/// Full information on the used extended public key: fingerprint of the
/// master extended public key and a derivation path from it.
pub type KeySource = (Fingerprint, DerivationPath);
'''

We could make KeySource a newtype, but in another pr... i dont see how to discard the conversion without  changing the KeySource itself. 

Ill gladly work on it after this one. by now, i would leave the example as it is.

Reply:
Oh, I think I understand now. OK, fine to look at it later. It may be that we can't actually change it because it might break the spec but we'd have to dig much deeper.